nav2_map_server/include/nav2_map_server/map_2d/map_server_2d.hpp

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) 2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_MAP_SERVER__MAP_2D__MAP_SERVER_2D_HPP_
+#define NAV2_MAP_SERVER__MAP_2D__MAP_SERVER_2D_HPP_
+
+#include <string>
+#include <memory>
+
+#include "nav_msgs/msg/occupancy_grid.hpp"
+#include "nav_msgs/srv/get_map.hpp"
+#include "nav2_msgs/srv/load_map.hpp"
+
+#include "nav2_map_server/map_server_core.hpp"
+
+namespace nav2_map_server
+{
+
+/**
+ * @class nav2_map_server::MapServer<nav_msgs::msg::OccupancyGrid>
+ * @brief Parses the map yaml file and creates a service and a publisher that
+ * provides occupancy grid, hosts GetMap, and LoadMap services.
+ * GetMap service default name : "map"
+ * LoadMap service default name : "load_map"
+ */
+template<>
+class MapServer<nav_msgs::msg::OccupancyGrid>: public nav2_util::LifecycleNode
```

Comment:
Same here, the templating seems very overkill, this is named `map_server_2d` in the `map_2d` directory, there's no expectation by that naming for this to support anything other than OccGrids and there's no template types defined to override even if there was a usecase for another type